modules/shared/src/main/scala/org/tessellation/scalacheck/generators/trust.scala

```diff
@@ -0,0 +1,64 @@
+package org.tessellation.scalacheck.generators
+
+import org.tessellation.schema.TrustValueRefinement.TrustValueRefinement
+import org.tessellation.schema.peer.PeerId
+import org.tessellation.schema.trust._
+import org.tessellation.security.hex.Hex
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.refineV
+import org.scalacheck.Arbitrary.arbitrary
+import org.scalacheck.{Arbitrary, Gen}
+
+object trust {
+
+  implicit val arbitraryHexString: Arbitrary[String] = Arbitrary {
+    Gen.choose(0, 0xffffff).flatMap(_.toHexString)
+  }
+
+  implicit val arbitraryTrustDouble: Arbitrary[Double] = Arbitrary {
+    Gen.choose(-1.0, 1.0)
+  }
+
+  implicit val arbitraryOptionalTrustValue: Arbitrary[Option[Double]] = Arbitrary {
+    Gen.option(Gen.choose(-1.0, 1.0))
+  }
+
+  val genHex = arbitrary[String].flatMap(Hex(_))
+
+  val genScore = arbitrary[Double]
+    .flatMap(refineV[TrustValueRefinement](_))
+    .flatMap(_ match {
+      case Left(_)      => Option[Score](Score(0.0))
+      case Right(value) => Option[Score](Score(value))
+    })
+
+  val genRating = arbitrary[Double]
+    .flatMap(refineV[TrustValueRefinement](_))
+    .flatMap(_ match {
+      case Left(_)      => Option[Rating](Rating(0.0))
+      case Right(value) => Option[Rating](Rating(value))
+    })
+
+  val genObservationAdjustment = arbitrary[Double]
+    .flatMap(refineV[TrustValueRefinement](_))
+    .flatMap(_ match {
+      case Left(_)      => Option[ObservationAdjustment](ObservationAdjustment(0.0))
+      case Right(value) => Option[ObservationAdjustment](ObservationAdjustment(value))
+    })
+
+  val genPeerId = genHex.flatMap(PeerId(_))
```

Comment:
the arbitrary for peer id already exists, please reuse it (abstract if needed)

Reply:
Changed.